File: 102/102_version2.java
class Solution {
    List<List<Integer>> res = new ArrayList<>();
    public List<List<Integer>> levelOrder(TreeNode root) {
        dfs(root,0);
        return res;
    }
    
    private void dfs(TreeNode node,int level){
        if(node!=null){
            if(level==res.size()) res.add(new ArrayList<>());
            res.get(level).add(node.val);
            dfs(node.left,level+1);
            dfs(node.right,level+1);
        }
    }
}

File: 102/102_version2.py
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        self.res = []
        self.dfs(root,0)
        return self.res
    def dfs(self,node,level):
        if node:
            if level == len(self.res):
                self.res.append([])
            self.res[level].append(node.val)
            self.dfs(node.left,level+1)
            self.dfs(node.right,level+1)
